presto-main/src/main/java/io/prestosql/metadata/AggregationFunctionMetadata.java

```diff
@@ -13,18 +13,39 @@
  */
 package io.prestosql.metadata;
 
+import io.prestosql.spi.type.StandardTypes;
 import io.prestosql.spi.type.TypeSignature;
+import io.prestosql.spi.type.TypeSignatureParameter;
 
+import java.util.Arrays;
 import java.util.Optional;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
 public class AggregationFunctionMetadata
 {
     private final boolean orderSensitive;
     private final Optional<TypeSignature> intermediateType;
 
+    public AggregationFunctionMetadata(boolean orderSensitive, TypeSignature... intermediateTypes)
```

Comment:
I think it'd be cleaner, more explicit, and symmetric (what gets passed here would match what gets returned by `getIntermediateType()` below) to take a single type here and make the caller build a row type if necessary. That could be done with a helper function on the caller side.

Reply:
This is a setup for future work